packages/graphql/src/translate/queryAST/factory/FilterFactory.ts

```diff
@@ -613,37 +602,63 @@ export class FilterFactory {
         }
     }
 
+    private convertRelationshipOperatorToLegacyOperator(operator: string): RelationshipWhereOperator {
+        switch (operator) {
+            case "some":
+                return "SOME";
+            case "all":
+                return "ALL";
+            case "single":
+                return "SINGLE";
+            case "none":
+                return "NONE";
+        }
+        throw new Error(`Invalid operator ${operator}`);
+    }
+
     private createRelatedNodeFilters({
         relationship,
         value,
         operator,
-        isNot,
+
         isConnection,
         isAggregate,
     }: {
         relationship: RelationshipAdapter;
-        value: any;
+        value: Record<string, any>;
         operator: FilterOperator | undefined;
-        isNot: boolean;
+
         isConnection: boolean;
         isAggregate: boolean;
     }): Filter | Filter[] {
         if (isAggregate) {
             return this.createAggregationFilter(relationship, value as AggregateWhereInput);
         }
-        if (operator && !isRelationshipOperator(operator)) {
+        if (!operator) {
+            const objectEntries = Object.entries(value);
+
+            if (objectEntries.length !== 1) {
+                throw new Error("Expected one quantifier in a relationship predicate");
+            }
+            const [genericOperator, genericValue] = objectEntries[0] as [string, any];
```

Comment:
I'm not sure `genericOperator` is a very good naming convention

We've fallen to that convention for the schema generation types, but here is a bit weird tbh. We are parsing an operator after all. Same with value

Reply:
I agree, same as the  comment above, at this moment is named generic just to differentiate from the deprecated syntax, in the future this will be just operator!